src/js/angular/core/directives/yasgui-component/yasgui-component.directive.js

```diff
@@ -0,0 +1,587 @@
+            };
+
+            const querySavedHandler = (successMessage) => {
+                toastr.success(successMessage);
+                $scope.savedQueryConfig = merge({}, $scope.savedQueryConfig, {
+                    saveSuccess: true
+                });
+            };
+
+            const autocompleteLocalNames = (term, canceler) => {
+                return AutocompleteRestService.getAutocompleteSuggestions(term, canceler.promise)
+                    .then(function (results) {
+                        canceler = null;
+                        return results.data;
+                    });
+            };
+
+            const getRepositoryStatementsCount = () => {
+                // A promise is returned because the $http of  angularjs use HttpPromise and its behavior is different than we expect.
+                // Here is an article that describes the problems AngularJS HttpPromise methods break promise chain {@link https://medium.com/@ExplosionPills/angularjs-httppromise-methods-break-promise-chain-950c85fa1fe7}
+                return RDF4JRepositoriesRestService.getRepositorySize($repositories.getActiveRepository())
+                    .then((response) => parseInt(response.data))
+                    .catch(function (data) {
+                        const params = {
+                            repo: $repositories.getActiveRepository(),
+                            error: getError(data)
+                        };
+                        toastr.warning($translate.instant('query.editor.repo.size.error', params));
+                    });
+            };
+
+            const onQueryAborted = (req) => {
+                if (req) {
+                    const repository = req.url.substring(req.url.lastIndexOf('/') + 1);
+                    const currentTrackAlias = req.header['X-GraphDB-Track-Alias'];
+                    return MonitoringRestService.deleteQuery(currentTrackAlias, repository);
+                }
+            };
+
+            const createParameter = (key, value) => {
+                return {key, value};
+            };
+
+            const getCommandParameters = (response) => {
+                return [createParameter('name', response.data.name)];
+            };
+
+            const toNoCommandResponse = () => {
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.SUCCESS);
+            };
+
+            const toHasNotSupport = (response) => {
+                const parameters = [
+                    createParameter('connectorName', response.data.connectorName),
+                    createParameter('pluginName', response.data.pluginName)
+                ];
+                return new BeforeUpdateQueryResult(BeforeUpdateQueryResultStatus.ERROR, 'query.editor.inactive.plugin.warning.msg', parameters);
+            };
+
+            const createConnectorProgressDialog = (actionName, iri, connectorName) => {
```

Comment:
IMO this also doesn't belong to this directive. It's better to move it in the respective view if possible

Reply:
I will do it in next MR.